addon/ng2/tasks/build-watch.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var chalk    = require('chalk');
+var Task     = require('ember-cli/lib/models/task');
+var Watcher  = require('ember-cli/lib/models/watcher');
+var Builder  = require('ember-cli/lib/models/builder');
+var Promise  = require('ember-cli/lib/ext/promise');
+
+module.exports = Task.extend({
+  run: function(options) {
+    this.ui.startProgress(
+      chalk.green('Building'), chalk.green('.')
+    );
+    
+    var watcher = new Watcher({
+      ui: this.ui,
+      builder: new Builder({
+        ui: this.ui,
+        outputPath: options.outputPath,
+        environment: options.environment,
+        project: this.project
+      }),
+      analytics: this.analytics,
+      options: options
+    });
+    
+    return {
+      watcher: watcher,
+      completion: watcher.then(function() {
+        return new Promise(function () {}); // Run until failure or signal to exit
```

Comment:
This feels wrong to me, but I don't know enough of Watcher to know if that's a common pattern. Could you just confirm in a comment that this promise should never resolve, with a link somewhere (ember github?) where people can refer instead of filling PRs/issues?